pandas/plotting/_matplotlib/core.py

```diff
@@ -409,13 +411,19 @@ def _compute_plot_data(self):
         if is_empty:
             raise TypeError("no numeric data to plot")
 
-        # GH25587: cast ExtensionArray of pandas (IntegerArray, etc.) to
-        # np.ndarray before plot.
-        numeric_data = numeric_data.copy()
-        for col in numeric_data:
-            numeric_data[col] = np.asarray(numeric_data[col])
+        def convert_to_ndarray(data):
+            # GH32073: cast to float if values contain nulled integers
+            if is_integer_dtype(data.dtype) and is_extension_array_dtype(data.dtype):
+                return data.to_numpy(dtype="float", na_value=np.nan)
 
-        self.data = numeric_data
+            # GH25587: cast ExtensionArray of pandas (IntegerArray, etc.) to
```

Comment:
why is this not just `np.asarray(data)` ?

Reply:
That will result in object dtype with NAs, which matplotlib doesn't understand.